### PR TITLE
Upgrade autocomplete library

### LINKED ISSUE #2490: Drop jquery.autocomplete plugin and use jquery ui
The version of jquery.autocomplete we use is outdated, and is available in jQuery UI. It is blocking us upgrading jQuery.

jQuery autocomplete is imported inside index.js
```
import '../../../../vendor/js/jquery-autocomplete/jquery.autocomplete-modified.js';
```
We also include jQuery UI which has autocomplete functionality using:
```
$.ui.autocomplete
```

- [ ] Drop jQuery autocomplete vendor import
- [ ] Update existing usages of the autocomplete function inside openlibrary/plugins/openlibrary/js/autocomplete.js to use $.ui.autocomplete function

### Evidence / Screenshot (if possible)

When sharing your pull request make sure to test the search box in the top right corner and detail what you needed to change to make use of jQuery.Ui

### Expectation
Before and after your pull request it will be possible to search

### Stakeholders
@cdrini

### PR DESCRIPTION
**Pull request by jdlrobson**
Closes #2490

Drops the modified jquery autocomplete and uses the more modern jquery ui library.

Note the main search box in the top right doesn't use jquery autocomplete so is not impacted by this change.

### Testing

Compare jquery autocomplete experience on authors/languages/works fields with production. They should more or less look the same.

### Evidence

Screen Shot 2021-01-29 at 11 57 27 AM

### Stakeholders
@cdrini 
maybe we can merge some of the earlier ones by pulling out separately e.g.  eadc7f2  and  299dadf  if code review is time consuming. Let me know.

### CODE REVIEW

**Comment by jdlrobson** [comment]
Bump. I guess this is my top priority right now given we want to upgrade jquery version.

**Comment by tfmorris** [comment]
> I couldn't get the highlighting to work with the new library so I'd like to suggest dropping it.

Without highlighting, how does the user know which item in the list they are selecting? (or is that comment obsolete?)

**Comment by jdlrobson** [comment]
> > I couldn't get the highlighting to work with the new library so I'd like to suggest dropping it.
> 
> Without highlighting, how does the user know which item in the list they are selecting? (or is that comment obsolete?)

it's still relevant I'm afraid.

Looks like this patch might be defunct now. I notice @cdrini made some changes in 04b5bc032b which make pulling out the JS from inline HTML even harder so maybe I should just focus on 
7ba0b8f  for the time being.

**Comment by jdlrobson** [comment]
Okay I've broken out #3279 and #3278  - hopefully quick reviews.

**Comment by jdlrobson** [comment]
@cdrini @jamesachamp would it be helpful for me to rebase this now given the work with moving jquery ui out of vendor ? This is the only thing as I recall that blocks us upgrading our version of jQuery and dropping jquery migrate.

**Comment by cclauss** [comment]
~#3278~ is merged and ~#3279~ is closed.  Please resolve conflicts or close.

**Comment by jdlrobson** [comment]
@cclauss  see my comment on https://github.com/internetarchive/openlibrary/pull/2734#issuecomment-760544733
I'd love to rebase this but I'm waiting for a reply - I don't want to waste my time if someone can't commit to reviewing it, particularly given this part of the codebase is in flux right now.

**Comment by jamesachamp** [comment]
@jdlrobson sorry for the delay.  It looks like this will resolve the errors thrown by `setup_multi_input_autocomplete` when jquery ui's autocomplete is used.  If that's true, I think that it's worthwhile to rebase and merge this pull request.

**Comment by jdlrobson** [comment]
@jamesachamp rebased. The patch is a lot better from the redo and the bundle size a lot smaller. I've descoped the removal of inline scripts for now as I have another issue to deal with that and it keeps this pull request light.

Let me know if you have any questions.

File: openlibrary/plugins/openlibrary/js/autocomplete.js [hunk 1]
```diff
@@ -16,59 +16,92 @@ export default function($) {
...
-            .nomatch(function(){
-                $(`#${this.id}-key`).val('');
-                $(this).addClass('reject');
-            })
-            .keypress(function() {
+                    // When no results if callback is defined, append a create new entry
+                    if (!results.length &&
+                        (
+                            ol_ac_opts.addnew ||
+                            (ol_ac_opts.addnew && ol_ac_opts.addnew(term))
```
Comment (jamesachamp) [inline openlibrary/plugins/openlibrary/js/autocomplete.js:84]: I don't quite understand what's happening with this expression.  If `addnew` is defined, won't this always return true regardless of what `ol_ac_opts.addnew(term)` returns?

Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/autocomplete.js:84]: I don't understand either :) Generally when I port code, I'm pretty conservative to retain existing logic where ever possible. This is the code that was originally on L41 just in a different place:

https://github.com/internetarchive/openlibrary/blob/a61250b4ade345430e406d99eb04ae102713899a/openlibrary/plugins/openlibrary/js/autocomplete.js#L41

The `parse` option is no longer supported for the jquery ui autocomplete, so this instead needs to run as part of the `source` option.

Once we've got onto jquery UI that means we can push ahead with upgrading the jquery version, which will mean writing lots of unit tests for things like this.

That make sense?

Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/autocomplete.js:84]: And now as I write this I see I missed the boolean check when porting which does make this make more sense :)

Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/autocomplete.js]: Now:
```
ol_ac_opts.addnew === true ||
                            (ol_ac_opts.addnew && ol_ac_opts.addnew(term))
                            ```

**Comment by jdlrobson** [comment]
@jamesachamp  can you take a look? I'd like to push ahead with fixing up my other pull requests on Friday when I have some free time but they are blocked on this.

**Review by jamesachamp — APPROVED** [review]

**Comment by jamesachamp** [comment]
This all looks good to me, and runs properly when tested locally.  I'm not able to merge branches (nor should I be, given my level of experience).  @cdrini could you take a look at this when you are able?

**Comment by mekarpeles** [comment]
I think these changes should be good to land, adding to staging then will merge later today.

**Comment by mekarpeles** [comment]
Works for me on staging, merging
_home_mek_2021-02-11-224023_1600x900_scrot png

**Pull request merged**